app.rb

```diff
@@ -0,0 +1,155 @@
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid selection'
+    end
+  end
+
+  def create_student
+    puts 'Create a new student'
+    print 'Student Name:'
+    name = gets.chomp
+    print 'Student Age: '
+    age = gets.chomp.to_i
+    print 'Has parent permission? [y/n]: '
+    parent_permission = gets.chomp.downcase
+
+    case parent_permission
+    when 'n'
+      puts 'Student doesn\'t have parent\'s permission, unable to create student.'
+    when 'y'
+      student = Student.new(@main_classroom, age, name, parent_permission: parent_permission)
+      @people << student
+      puts 'Student created successfully!'
+    end
+  end
+
+  def create_teacher
+    puts 'Create a new teacher'
+    print 'Teacher Name: '
+    name = gets.chomp
+    print 'Teacher Age: '
+    age = gets.chomp.to_i
+    print 'Teacher\'s specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name)
+    @people << teacher
+    puts 'Teacher created successfully!'
+  end
+
+  def create_book
+    puts 'Create a new book.'
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts "Book #{title} created successfully"
+  end
+
+  def create_rental
+    puts 'Select a book using it\'s number: '
+    @books.each_with_index { |book, index| puts "#{index}) #{book.title}" }
+    book_id = gets.chomp.to_i
+    book = @books[book_id]
+
+    puts 'Select a person using it\'s number: '
+    @people.each_with_index { |person, index| puts "#{index}) #{person}" }
+    person_id = gets.chomp.to_i
+    person = @people[person_id]
+
+    puts 'Date: '
+    date = gets.chomp
+    @rentals << Rental.new(date, person, book)
+    puts 'Records created successfully'
+  end
+
+  def list_all_rentals
+    puts 'Enter ID of person: '
+    id = gets.chomp.to_i
+    rentals = @rentals.select { |rent| rent.person.id == id }
+    puts 'Rentals:'
+    puts rentals
+    # rentals = @rentals.select |r|
+    # if r.owner.id == id
+    #   puts 'Rentals: '
+    #   puts rentals
+    # else
+    #   puts 'Records not found for given ID'
+    # end
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+
+  def execute_inputs
+    input = Integer(gets.chomp)
+    case input
+    when 1 then list_books
+    when 2 then list_people
+    when 3 then create_person
+    when 4 then create_book
+    when 5 then create_rental
+    when 6 then list_all_rentals
+    when 7
+      puts 'Thank you for using this app. Goodbye'
+      exit
+    else
+      puts 'Kindly enter a number between 1-7'
+    end
+  end
+
+  # rubocop:enable Metrics/CyclomaticComplexity
+
+  def start
+    puts 'Welcome to School Library App!'
+    puts ' '
+    puts 'Please choose an option by entering a number:'
+    loop do
+      puts '1 - List all books'
+      puts '2 - List all people'
+      puts '3 - Create a person'
+      puts '4 - Create a book'
```

Comment:
- [ ] I suggest you move these functions to the ```main.rb``` file, this will ensure separation of functionality to make your ```main.rb``` file deal with user inputs and the ```app.rb``` process the data from the inputs.